components/dashboard/src/utils.ts

```diff
@@ -213,3 +213,22 @@ export class ReplayableEventEmitter<EventTypes extends EventMap> extends EventEm
         return this.reachedEnd;
     }
 }
+
+export function isTrustedUrlOrPath(urlOrPath: string) {
+    try {
+        const ok = URL.canParse(urlOrPath);
```

Comment:
We've recently had issues with users not being on up-to-date browsers and therefore not having the `URL.canParse()` static method. It does not look like babel wants to translate this, so it might be a good idea to use something like https://github.com/jsdom/whatwg-url here until we can reach >= 95% of users.

Additionally, `URL.parse()` shipped very recently, which returns `URL | null`, but that isn't even in the aforementioned polyfill yet. Made https://github.com/jsdom/whatwg-url/issues/277 for that

Reply:
Addressed https://github.com/gitpod-io/gitpod/pull/20152/commits/edfabf88a382e75eae93f5b0ff04ae6902f994e2